test/RevokableTokenLock.spec.ts

```diff
@@ -0,0 +1,113 @@
+  describe('#setRevoker', async () => {
+    it('should revert if caller is not owner', async () => {
+      await expect(revokableTokenLock.connect(recipient).setRevoker(other.address)).to.be.revertedWith(
+        'Ownable: caller is not the owner'
+      );
+    });
+    it('should revert if _revoker is the null address', async () => {
+      await expect(revokableTokenLock.setRevoker(ZERO_ADDRESS)).to.be.revertedWith('RevokableTokenLock: null address');
+    });
+    it('should not revert if caller is owner and _revoker is not the null address', async () => {
+      await expect(revokableTokenLock.setRevoker(other.address)).to.not.be.reverted;
+    });
+    it('should update value of revoker in contract to other address', async () => {
+      await revokableTokenLock.setRevoker(other.address);
+      const revoker = await revokableTokenLock.revoker();
+      expect(revoker).to.equal(other.address);
+    });
+  });
+  describe('#revoke', async () => {
+    const amount = ethers.utils.parseEther('20');
+    let transferred: BigNumber;
+    let remaining: BigNumber;
+    let unlockBegin: BigNumber;
+    let unlockCliff: BigNumber;
+    let unlockEnd: BigNumber;
+    let claimedAmounts: BigNumber;
+    let lockedAmounts: BigNumber;
+    before('Set up data', async () => {
+      ({unlockBegin, unlockCliff, unlockEnd} = await revokableTokenLock.vesting(recipient.address));
+      transferred = amount.mul(unlockCliff.sub(unlockBegin)).div(unlockEnd.sub(unlockBegin));
+      remaining = amount.sub(transferred);
+    });
+    it('should revert if caller is not owner or revoker', async () => {
+      await expect(revokableTokenLock.connect(other).revoke(recipient.address)).to.be.revertedWith(
+        'RevokableTokenLock: onlyAuthorizedActors'
+      );
+    });
+    it('should not revert if caller is revoker', async () => {
+      await expect(revokableTokenLock.connect(revoker).revoke(recipient.address)).to.not.be.reverted;
+    });
+    it('should not revert if caller is owner', async () => {
+      await expect(revokableTokenLock.revoke(recipient.address)).to.not.be.reverted;
+    });
+    it('should claim any tokens vested and revoke the rest when revoke is called', async () => {
+      // TODO: how to move this into a fixture?
+      await token.approve(revokableTokenLock.address, amount);
+      await revokableTokenLock.lock(recipient.address, amount);
+      await setNextBlockTimestamp(unlockCliff.toNumber());
+
+      // Call revoke and check token balances
+      await expect(() => revokableTokenLock.revoke(recipient.address)).to.changeTokenBalances(
+        token,
+        [recipient, owner],
+        [transferred, remaining]
+      );
+
+      // Check contract values
+      ({unlockEnd, claimedAmounts, lockedAmounts} = await revokableTokenLock.vesting(recipient.address));
+      expect(claimedAmounts).to.equal(transferred);
+      expect(lockedAmounts).to.equal(claimedAmounts);
+      expect(unlockEnd).to.equal(unlockCliff);
```

Comment:
I think the test should check fewer implementation details and more functionality:
1) the claimer's balance increased by `amounts`
2) the owner's balance increased by the `remaining`
3) if you keep moving forward in time and check `claimableBalance`, it should return 0

Reply:
1 and 2 are actually checked on lines 88-92!

Will add test for 3.